libs/blocks/global-navigation/global-navigation.js

```diff
@@ -236,7 +238,8 @@ class Gnav {
         ${breadcrumbs}
       </div>`;
 
-    this.el.append(this.elements.curtain, this.elements.aside, this.elements.topnavWrapper);
+    if (this.useFederatedContent) federatePictureSources(this.elements.topnavWrapper);
```

Comment:
self-review: As the pictureSources are already being federated right after fetching, this should not be needed anymore.